### PR TITLE
3 invoices

### PR DESCRIPTION
**Pull request by Scott-Borecki**
### What I Did:
- Completed the following user stories:
  - #8 
  - #9 
  - #10 
- Refactored code styling and base repo tests along the way.
- Updated factories and test setup for more robust testing.

### Road Blocks
- Extracting BulkDiscount ID from AR object with multiple aggregations.  Ended up breaking this out into helper methods.

### Test Coverage
- [X] I have tested this code with RSpec
- Still need to go through base user stories to get full coverage.
- Edge cases that need testing:
  - Need to test for cases where the invoice has items from two merchants.
  - Need to test for cases where the invoice has duplicate items

### Questions/Notes:
- Maybe add a method that gets all invoice items for an invoice and combines duplicate items into a single line item and adds quantities together.  This would be used in the display.  
- Alternatively, could check if there are any duplicate items and then apply the discount based on the total quantity of that item on the invoice.  This would preserve the invoice item rows in the table so might be preferred.

### CODE REVIEW

File: lib/tasks/import.rake [hunk 1]
```diff
@@ -1,27 +1,37 @@
 require 'csv'
+require 'faker'
 
 task :import, [:customers] => :environment do
+  ActiveRecord::Base.connection.reset_pk_sequence!('customers')
   CSV.foreach('db/data/customers.csv', headers: true) do |row|
-    Customer.create!(row.to_hash)
+    Customer.create!({ id:          row[0],
```
[the remark this answers is not in the document]
Comment (Scott-Borecki) [reply lib/tasks/import.rake]: Updated
